ArcGISRuntimeSDKQt_CppSamples/Geometry/ConvexHull/ConvexHull.cpp

```diff
@@ -0,0 +1,135 @@
+// [WriteFile Name=ConvexHull, Category=Geometry]
+// [Legal]
+// Copyright 2020 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ConvexHull.h"
+
+#include "GeometryEngine.h"
+#include "Graphic.h"
+#include "GraphicsOverlay.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MultipointBuilder.h"
+#include "PointCollection.h"
+#include "SimpleFillSymbol.h"
+#include "SimpleMarkerSymbol.h"
+
+using namespace Esri::ArcGISRuntime;
+
+ConvexHull::ConvexHull(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::topographic(this), this))
+{
+  setupGraphics();
+}
+
+ConvexHull::~ConvexHull() = default;
+
+void ConvexHull::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ConvexHull>("Esri.Samples", 1, 0, "ConvexHullSample");
+}
+
+MapQuickView* ConvexHull::mapView() const
+{
+  return m_mapView;
+}
+
+void ConvexHull::displayConvexHull()
+{
+  Geometry convexHull = GeometryEngine::convexHull(m_inputsGraphic->geometry());
+
+  // change the symbol based on the returned geometry type
+  if (convexHull.geometryType() == GeometryType::Point)
+  {
+    m_convexHullGraphic->setSymbol(m_markerSymbol);
+  }
+  else if (convexHull.geometryType() == GeometryType::Polyline)
+  {
+    m_convexHullGraphic->setSymbol(m_lineSymbol);
+  }
+  else if (convexHull.geometryType() == GeometryType::Polygon)
+  {
+    m_convexHullGraphic->setSymbol(m_fillSymbol);
+  }
+
```

Comment:
It might be worth adding a final `else { }` block here with some `qWarn` message.